lib/Doctrine/Common/Cache/PhpFileCache.php

```diff
@@ -101,8 +93,14 @@ protected function doSave($id, $data, $lifeTime = 0)
             'data'      => $data
         );
 
-        $value  = var_export($value, true);
-        $code   = sprintf('<?php return %s;', $value);
+        $serializeValue = serialize($value);
+
+        if (strpos($serializeValue, 'r:') !== false || (is_object($data) && !method_exists($data, '__set_state'))) {
```

Comment:
Before the modification I made the PhpFileCache was throwing an exeption when $data was an object without method __set_state(). 

The modification I made trigger when we can't var_export $data (redondant reference or method __set_state() not implemented) and serialize it.